ppl/lake/tsdirstream.go

```diff
@@ -0,0 +1,120 @@
+package lake
+
+import (
+	"container/heap"
+	"context"
+	"sort"
+
+	"github.com/brimsec/zq/ppl/lake/chunk"
+	"github.com/brimsec/zq/zbuf"
+	"golang.org/x/sync/errgroup"
+)
+
+type tsDirStream struct {
+	ch  chan tsDirStreamResult
+	err error
+}
+
+func newTsDirStream(ctx context.Context, lk *Lake, tsDirs []tsDir) *tsDirStream {
+	sort.Slice(tsDirs, func(i, j int) bool {
+		if lk.DataOrder == zbuf.OrderAsc {
+			return tsDirs[i].Ts < tsDirs[j].Ts
+		}
+		return tsDirs[j].Ts < tsDirs[i].Ts
+	})
+
+	t := &tsDirStream{ch: make(chan tsDirStreamResult)}
+	go t.run(ctx, lk, tsDirs)
+	return t
+}
+
+func (t *tsDirStream) run(ctx context.Context, lk *Lake, tsDirs []tsDir) {
+	ch := make(chan tsDirStreamResult)
+	results := &tsDirStreamResultHeap{order: lk.DataOrder}
```

Comment:
Simplify by using a channel of channels instead of `tsDirStreamResultHeap`?

Reply:
Interesting.

Reply:
@nwt is this what you are talking about: https://github.com/brimsec/zq/pull/2162/commits/8daf603c6883973472396131ddf83d1127a63f7d